tests/tezos-modules.nix

```diff
@@ -0,0 +1,51 @@
+# SPDX-FileCopyrightText: 2021 TQ Tezos <https://tqtezos.com/>
+#
+# SPDX-License-Identifier: LicenseRef-MIT-TQ
+let
+  nixpkgs = (import ../nix/nix/sources.nix).nixpkgs;
+  pkgs = import ../nix/build/pkgs.nix {};
+in import "${nixpkgs}/nixos/tests/make-test-python.nix" ({ ... }:
+{
+  machine = { ... }: {
+    nixpkgs.pkgs = pkgs;
+    imports = [ ../nix/modules/tezos-node.nix
+                ../nix/modules/tezos-signer.nix
+                ../nix/modules/tezos-accuser.nix
+                ../nix/modules/tezos-baker.nix
+                ../nix/modules/tezos-endorser.nix
+              ];
+
+    services.tezos-node.instances.edonet.enable = true;
+
+    services.tezos-signer.instances.edonet = {
+      enable = true;
+      networkProtocol = "http";
+    };
+
+    services.tezos-accuser.instances.edonet = {
+      enable = true;
+      baseProtocol = "008-PtEdo2Zk";
+    };
+
+    services.tezos-baker.instances.edonet = {
+      enable = true;
+      baseProtocol = "008-PtEdo2Zk";
+    };
+
+    services.tezos-endorser.instances.edonet = {
+      enable = true;
+      baseProtocol = "008-PtEdo2Zk";
+    };
+
+  };
+
+  testScript = ''
+    start_all()
+
+    machine.wait_for_unit("tezos-edonet-tezos-node.service")
+    machine.wait_for_unit("tezos-edonet-tezos-signer.service")
+    machine.wait_for_unit("tezos-edonet-tezos-accuser.service")
+    machine.wait_for_unit("tezos-edonet-tezos-baker.service")
+    machine.wait_for_unit("tezos-edonet-tezos-endorser.service")
+  '';
```

Comment:
Needs moar test IMO. This will pass even if the services break after a second of running. Would be nice to see some sanity checks here.

Reply:
I'm not sure how to test more precisely than what I do in the latest version, as no service other than `tezos-node` really has an rpc interface. Hopefully just checking status is enough.